src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,161 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private final int defaultSize = 5;
+    private Object[] defaultNewCapacity;
+    private T[] elementArray;
+    private int size;
+
+    public ArrayList() {
+        elementArray = (T[]) new Object[defaultSize];
+    }
+
+    public ArrayList(int capacity) {
+        elementArray = (T[]) new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        grow();
+        elementArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException(" Index "
+                    + index + " out of bounds for length " + size);
+        }
+        defaultNewCapacity = new Object[size + 1];
+        for (int i = 0; i < defaultNewCapacity.length; i++) {
+            if (i < index) {
+                defaultNewCapacity[i] = elementArray[i];
+            }
+            if (i == index) {
+                defaultNewCapacity[i] = value;
+            }
+            if (i > index) {
+                defaultNewCapacity[i] = elementArray[i - 1];
+            }
+        }
```

Comment:
This also applies to this case
<img width="491" alt="Screen Shot 2022-10-22 at 1 36 59 PM" src="https://user-images.githubusercontent.com/90692308/197334661-b9d16f4c-f388-4506-a1e4-3d31ee23b283.png">